kinto_client/__init__.py

```
@@ -50,43 +48,38 @@ def create_session(server_url=None, auth=None, session=None):
 
 
 class Endpoints(object):
+    endpoints = {
+        'root':         '{root}/',
+        'batch':        '{root}/batch',
+        'buckets':      '{root}/buckets',
+        'bucket':       '{root}/buckets/{bucket}',
+        'collections':  '{root}/buckets/{bucket}/collections',
+        'collection':   '{root}/buckets/{bucket}/collections/{collection}',
+        'records':      '{root}/buckets/{bucket}/collections/{collection}/records',      # NOQA
+        'record':       '{root}/buckets/{bucket}/collections/{collection}/records/{id}'  # NOQA
+    }
+
     def __init__(self, root=''):
         self._root = root
 
-    def root(self):
-        return '{root}/'.format(root=self._root)
-
-    def buckets(self):
-        return '{root}/buckets'.format(root=self._root)
-
-    def bucket(self, bucket):
-        return '{root}/buckets/{bucket}'.format(root=self._root, bucket=bucket)
-
-    def collections(self, bucket):
-        return ('{root}/buckets/{bucket}/collections'
-                .format(root=self._root, bucket=bucket))
-
-    def collection(self, bucket, coll):
-        return ('{root}/buckets/{bucket}/collections/{coll}'
-                .format(root=self._root, bucket=bucket, coll=coll))
-
-    def records(self, bucket, coll):
-        return ('{root}/buckets/{bucket}/collections/{coll}/records'
-                .format(root=self._root, bucket=bucket, coll=coll))
-
-    def record(self, bucket, coll, record_id):
-        return ('{root}/buckets/{bucket}/collections/{coll}/records/{rid}'
-                .format(root=self._root, bucket=bucket, coll=coll,
-                        rid=record_id))
+    def get(self, endpoint, **kwargs):
+        # Remove nullable values from the kwargs, and slugify the values.
+        kwargs = dict((k, utils.slugify(v))
```

I would update the comment to add:

    # This is to make sure that ``bucket="my bucket"`` converts into ``bucket="my-bucket"``.

I like the slugify idea, also I wonder if we should allow it implicitely or let the client raise telling that this ID is invalid?

I can imagine kinto servers configured with ID policies that allows for spaces and unicode characters. (https://kinto.dev.mozaws.net/v1/buckets/見/collections/香港/records)